### PR TITLE
streamingccl: rename cluster settings

### PR DESCRIPTION
**Pull request by stevendanna**
This renames most cluster settings related to streaming replication to start with either physical_replication.producer or physical_replication.consumer.

Epic: none

Release note (enterprise change): Settings related to the private
preview physical replication feature have been renamed to be more
consistent. For example,
`bulkio.stream_ingestion.minimum_flush_interval` is now
`physical_replication.consumer.minimum_flush_interval`.

### CODE REVIEW

File: pkg/ccl/streamingccl/settings.go [hunk 1]
```diff
@@ -39,7 +40,20 @@ var StreamReplicationJobLivenessTimeout = settings.RegisterDurationSetting(
 	"stream_replication.job_liveness_timeout",
 	"controls how long we wait for to kill an inactive producer job",
 	3*24*time.Hour,
-	settings.WithName("stream_replication.job_liveness.timeout"),
+	settings.WithRetiredName("stream_replication.job_liveness.timeout"),
```
Comment (knz) [inline pkg/ccl/streamingccl/settings.go:43]: you don't need to register a retired name if it's the same as the key. maybe worth clarifying as a comment on the WithRetiredName API function.

**Review by knz — APPROVED** [review]
lgtm modulo removing a superfluous api call.

**Comment by knz** [comment]
i recommend a release note for the public settings.

File: pkg/ccl/streamingccl/settings.go (hunk 1 above)
Comment (stevendanna) [reply pkg/ccl/streamingccl/settings.go:43]: It's not the same as the key though: liveness_timeout vs liveness.timeout.

Comment (knz) [reply pkg/ccl/streamingccl/settings.go:43]: nice. carry on

**Review by knz — COMMENTED** [review]
> Reviewed 3 of 6 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/111197)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @dt, @rhu713, and @stevendanna)

___
*[`-- commits` line 8 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/111197#-Nfou7va3w3xGjapKKae):*
I'm curious; why not?

**Review by knz — APPROVED** [review]

**Comment by stevendanna** [comment]
bors r=knz

**Pull request merged**